compiler/p/codegen/PPCBinaryEncoding.cpp

```diff
@@ -902,12 +902,6 @@ uint8_t *TR::PPCAlignmentNopInstruction::generateBinaryEncoding()
 
    cg()->addAccumulatedInstructionLengthError(getEstimatedBinaryLength() - currentMisalign);
 
-   // When the trace log prints the list of instructions after binary encoding, we don't want this
```

Comment:
This `remove()` call was added for cosmetic reasons. Specifically, as the comment says, to prevent this alignment nop from showing up in tracefile output after it has expanded. While this is nice, it's not really necessary for the code to function correctly (and in fact, I've later realized that it can end up being more confusing than helpful sometimes).

However, one of the assumptions made by this code, namely that `remove()` leaves the `next` field alone so that binary encoding can continue, has now been violated. As a result, this no longer works correctly and can't be left as-is.